synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -91,3 +91,21 @@ impl<N: Network> Deref for ProvingKey<N> {
         &self.proving_key
     }
 }
+
+impl<N: Network> PartialEq for ProvingKey<N> {
+    fn eq(&self, other: &Self) -> bool {
+        self.deref() == other.deref()
+    }
+}
+
+impl<N: Network> Ord for ProvingKey<N> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.deref().cmp(other.deref())
```

Comment:
ditto

Reply:
Perhaps I could keep one of the automatic derivation, but when I remove some of them:
```
error[E0277]: the trait bound `N: Ord` is not satisfied
    --> synthesizer/process/src/trace/mod.rs:89:31
     |
89   |         self.transition_tasks.entry(proving_key).or_default().push(assignment);
     |                               ^^^^^ the trait `Ord` is not implemented for `N`
     |
     = note: required for `ProvingKey<N>` to implement `Ord`
```

Reply:
ugh, that's just typical :smile: